File: Java/02- Relacionando Classes/17- Produto/Advogado.java

package com.mycompany.mavenproject7;

public class Advogado extends Funcionario{
    private String oab;

    public Advogado(String oab, String cpf, String rg, String matricula, Setor setor, double salario, Sexo sexo, EstadoCivil estadocivil, String dataNascimento, String nome, String telefone, String email, Endereco endereco) {
        super(cpf, rg, matricula, setor, salario, sexo, estadocivil, dataNascimento, nome, telefone, email, endereco);
        this.oab = oab;
    }

    public String getOab() {
        return oab;
    }

    public void setOab(String oab) {
        this.oab = oab;
    }

    @Override
    public String toString() {
        return super.toString() +
               "\n=====Dados do Advogado=====" +
               "\nOAB: " + oab;
    }
    
}

File: Python/04- POO/03- Atividades Pontuada/01- Atividade Pontuada/models/enums/trabalho/advogado.py
from models.funcionario import Funcionario


class Advogado(Funcionario):
    def __init__(self, nome, telefone, email, endereco, cpf, rg, data_nascimento, sexo, matricula, setor, salario, oab: str) -> None:
        super().__init__(nome, telefone, email, endereco, cpf, rg, data_nascimento, sexo, matricula, setor, salario)
        self.oab = oab

    def salario_final(self):
        return self.salario

    def __str__(self) -> str:
        return (f"{super().__str__()}"
                f"\nOAB: {self.oab}"

               )